ui/src/components/Code/CodeBlock.tsx

```diff
@@ -60,8 +63,28 @@ export default function CodeBlock({ tabs }: CodeBlockProps) {
     setActiveTab(index);
   };
 
+  let isOutputTooLarge;
+  if (tabs[activeTab].content.length > maxRenderedOutputSizeBytes) {
+    isOutputTooLarge = true;
+  } else {
+    isOutputTooLarge = false;
+  }
+
+  const downloadJson = ({ content }) => {
+    const jsonContent = JSON.stringify(JSON.parse(content), null, 2); // Parse and stringify for pretty formatting
+    const blob = new Blob([jsonContent], { type: 'application/json' });
```

Comment:
If `context` is large then `JSON.parse` will crash Chrome. Also, there isn't a guarantee that `content` is valid JSON (it could even be HTML in some unhappy paths).

```suggestion
    const blob = new Blob([content], { type: 'application/json' });
```